src/main/java/space/obminyashka/items_exchange/controller/AuthController.java

```diff
@@ -114,6 +125,8 @@ public ResponseEntity<String> registerUser(@RequestBody @Valid UserRegistrationD
 
         if (userService.registerNewUser(userRegistrationDto)) {
             log.info("User with email: {} successfully registered", escapeHtml(userRegistrationDto.getEmail()));
+            token.setUser(userService.findByUsernameOrEmail(userRegistrationDto.getEmail()).get());
+            confirmationTokenRepository.save(token);
```

Comment:
You shouldn't invoke the repository directly from the controller - do it via service.